tests/unit/Store.ts

```diff
@@ -0,0 +1,251 @@
+			arguments: [ sortObject.property, sortObject.descending ],
+			normalizedArguments: [ [ sortObject ] ]
+		} ];
+		const expectedQueryLog2: any[] = [ {
+			type: 'sort',
+			arguments: [ sortObject ],
+			normalizedArguments: [ [ sortObject ] ]
+		} ];
+		const expectedQueryLog3: any[] = expectedQueryLog2.concat({
+			type: 'sort',
+			arguments: [ sortObjectArray ],
+			normalizedArguments: [ [ sortObject, lang.mixin({ descending: false }, sortObjectArray[1]) ] ]
+		});
+		const expectedQueryLog4 = expectedQueryLog3.concat({
+			type: 'sort', arguments: [ comparator ], normalizedArguments: [ comparator ]
+		});
+		let sortedCollection: dstore.Collection<any>;
+
+		sortedCollection = store.sort(sortObject.property, sortObject.descending);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog1);
+
+		sortedCollection = store.sort(sortObject);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog2);
+
+		sortedCollection = sortedCollection.sort(sortObjectArray);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog3);
+
+		sortedCollection = sortedCollection.sort(comparator);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog4);
+	},
+
+	'restore'() {
+		const store = new ConcreteStore({
+			Model: NonDeclareBasedModel
+		});
+		const restoredObject = (<any> store)._restore({ foo: 'original' });
+		assert.strictEqual(restoredObject.foo, 'original');
+		assert.strictEqual(restoredObject.restored, true);
+		assert.isTrue(restoredObject instanceof NonDeclareBasedModel);
+	},
+
+	events() {
+		const methodCalls: string[] = [],
+			events: string[] = [];
+
+		// rely on autoEventEmits
+		const store = new  ConcreteStore();
+		function countMethodCall(method: string, object: any) {
+			methodCalls.push(method);
+			return object;
+		}
+		store.put = countMethodCall.bind(null, 'put');
+		store.add = countMethodCall.bind(null, 'add');
+		store.remove = countMethodCall.bind(null, 'remove');
+
+		store.on('add', function (event: EventObject) {
+			events.push(event.type);
+		});
+		// test comma delimited as well
+		store.on('delete', function (event: EventObject) {
+			events.push(event.type);
+		});
+
+		store.on('update', function (event: EventObject) {
+			events.push(event.type);
+		});
+
+		store.put({});
+		store.add({});
+		store.remove(1);
+
+		assert.deepEqual(methodCalls, ['put', 'add', 'remove']);
+		assert.deepEqual.bind(this, events, ['update', 'add', 'delete']);
+	},
+
+	'events with beforeId'() {
+		const store = new ConcreteStore(),
+			beforeIds: Array<string | number> = [];
+
+		store.on('add', function (event) {
+			beforeIds.push(event.beforeId);
+		});
+
+		store.on('update', function (event) {
+			beforeIds.push(event.beforeId);
+		});
+
+		return store.add({}, <any> { beforeId: 123 }).then(function () {
+			return store.put({}, <any> { beforeId: 321 })
+		}).then( function () {
+			assert.deepEqual(beforeIds, [ 123, 321 ]);
+		});
+	},
+
+	'emit should catch errors thrown by listeners'() {
```

Comment:
I suspect this is not actually testing anything...  AFAIK we're no longer explicitly try/catching anywhere in the `emit` process, and neither does `Evented`.

Upon closer inspection, isn't `testEmit` set up wrong?  `emit` takes an object now, not a string.  Also, I'm not sure why `lateBind` is necessary here.